internal/exec/shell_utils.go

```diff
@@ -263,21 +279,24 @@ func execTerraformShellCommand(
 		}
 	}
 
+	// Merge env vars, ensuring componentEnvList takes precedence
+	mergedEnv := mergeEnvVars(os.Environ(), componentEnvList)
+
 	u.LogDebug(atmosConfig, "\nStarting a new interactive shell where you can execute all native Terraform commands (type 'exit' to go back)")
 	u.LogDebug(atmosConfig, fmt.Sprintf("Component: %s\n", component))
 	u.LogDebug(atmosConfig, fmt.Sprintf("Stack: %s\n", stack))
 	u.LogDebug(atmosConfig, fmt.Sprintf("Working directory: %s\n", workingDir))
 	u.LogDebug(atmosConfig, fmt.Sprintf("Terraform workspace: %s\n", workspaceName))
 	u.LogDebug(atmosConfig, "\nSetting the ENV vars in the shell:\n")
-	for _, v := range componentEnvList {
+	for _, v := range mergedEnv {
```

Comment:
Why is this needed when you've already _merged_ the TF var envs using the join?

Reply:
In order to include system env vars to the subshell (os.Environ()). Otherwise only the special cases (TF_* are included)

Reply:
previously this was handled with `append(os.Environ(), componentEnvList...),`, but I've moved it up so that we can see the final value with the debugging lines here

Reply:
> but I've moved it up so that we can see the final value with the debugging lines here

Still struggling to understand why...

why wouldn't this work then on line 282:

```
mergedEnv := append(os.Environ(), componentEnvList...)
```

Not seeing what your function adds in value.

Reply:
Is it that the ENV would be emitted twice? ... because `os.ProcAttr` ensures dupes are handled in precedence order. But output would show two. .. and you want to avoid the two outputs for the same env.

Reply:
If you dont include it, then we append the 2 lists and get both values:
```
Setting the ENV vars in the shell:

...

TF_CLI_ARGS_plan=-no-color                                    #<------- this
...
TF_CLI_ARGS_plan="-no-color -var-file=dev-station.terraform.tfvars.json"   #<------- and this
TF_CLI_ARGS_apply=-var-file=dev-station.terraform.tfvars.json
TF_CLI_ARGS_refresh=-var-file=dev-station.terraform.tfvars.json
TF_CLI_ARGS_import=-var-file=dev-station.terraform.tfvars.json
TF_CLI_ARGS_destroy=-var-file=dev-station.terraform.tfvars.json
TF_CLI_ARGS_console=-var-file=dev-station.terraform.tfvars.json
...
Starting process: /bin/zsh -l
```

The subshell sets the first value and ignores the second

Reply:
In `go`, this a list of strings. So `TF_CLI_ARGS_plan=-no-color` and `TF_CLI_ARGS_plan="-no-color -var-file=dev-station.terraform.tfvars.json"` are completely separate values and are not merged

Reply:
> Is it that the ENV would be emitted twice? ... because os.ProcAttr ensures dupes are handled in precedence order. But output would show two. .. and you want to avoid the two outputs for the same env.

Yes exactly. But `os.ProcAttr ensures dupes are handled in precedence order` is not true. The one defined first takes priority

Reply:
> The subshell sets the first value and ignores the second

Aha, didn't think that was the behavior. But if that's what you observe, it must be.